yaml-tests/src/main/java/com/apple/foundationdb/relational/yamltests/command/QueryCommand.java

```diff
@@ -130,28 +129,28 @@ private QueryCommand(int lineNumber, @Nonnull QueryInterpreter interpreter, @Non
                 "SkipConfig should not have gotten into QueryCommand " + lineNumber);
     }
 
-    public void execute(@Nonnull final RelationalConnection connection, boolean checkCache, @Nonnull QueryExecutor executor) {
+    public void execute(@Nonnull final YamlConnection connection, boolean checkCache, @Nonnull QueryExecutor executor) {
         try {
-            if (!(connection instanceof EmbeddedRelationalConnection) && checkCache) {
+            if (!connection.supportsMetricCollector() && checkCache) {
```

Comment:
Why `metricCollector` here?

Reply:
Because access to the `MetricCollector` is what we need in order to check the cache.
I originally had `supportsCheckCache`, but that felt bat from the vantage point of the `YamlConnection`, so I changed it to `supportsMetricCollector`, but that could probably use a comment.